mlflow/langchain/__init__.py

```diff
@@ -526,10 +526,17 @@ def load_retriever(persist_directory):
             ColSpec(type=DataType.string, name=input_key) for input_key in lc_model.input_keys
         ]
         input_schema = Schema(input_columns)
+
         output_columns = [
             ColSpec(type=DataType.string, name=output_key) for output_key in lc_model.output_keys
         ]
         output_schema = Schema(output_columns)
+
+        # TODO: empty output schema if multiple output_keys. fix later!
+        # https://databricks.atlassian.net/browse/ML-34706
+        if len(lc_model.output_keys) > 1:
+            output_schema = None
+
```

Comment:
if lc_model is a BaseRetriever, set the output schema to None.
BaseRetriever needs to return one column of serialized `List[Document]`. Currently, we serialize it to a string, which is consistent with the output schema.

After this PR, we serialize it to a list of dict, which is not supported by any output types, so we should set it to None for the same reason in ML-34706.

Can you also confirm that ML-34706 also handles the case where the column type is non-primitive?

Reply:
great catch!
the ticket mentions support for object types